app/lib/features/search/providers/pins.dart

```diff
@@ -0,0 +1,43 @@
+import 'package:acter/features/pins/providers/pins_provider.dart';
+import 'package:acter/features/search/model/base_delegate.dart';
+import 'package:acter/features/search/providers/search.dart';
+import 'package:atlas_icons/atlas_icons.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+class PinDetails extends BaseDelegate {
+  final Icon icon;
+  const PinDetails(String navigationTarget, String name, {required this.icon})
+      : super(navigationTarget: navigationTarget, name: name);
+}
+
+final AutoDisposeFutureProvider<List<PinDetails>> pinsFoundProvider =
+    FutureProvider.autoDispose((ref) async {
+  final pins = await ref.watch(pinsProvider.future);
+  final List<PinDetails> finalPins = [];
+  final searchValue = ref.watch(searchValueProvider).toLowerCase();
+
+  for (final pin in pins) {
+    final title = pin.title();
+    final pinId = pin.eventIdStr();
+    if (searchValue.isNotEmpty) {
+      if (!(title.toLowerCase()).contains(searchValue)) {
+        continue;
+      }
+    }
+    finalPins.add(
+      PinDetails(
+        '/pins/$pinId',
```

Comment:
no hardcoding of routes please.

Reply:
hmpf... I see you are only doing what was done here before ... so ... let's add a FIXME here and say that we want the base delegate to support proper route creation...
```suggestion
        // FIXME: it would be nicer to have proper route delegation rather than final strings here
        '/pins/$pinId',
```

Reply:
Ahh I forgot to change that, happens when you put your hands all over the code rather than actual fixation, will bring that in fresh PR :D